radixdlt-core/radixdlt/src/test/java/com/radixdlt/consensus/VoteTest.java

```diff
@@ -120,4 +120,39 @@ public void testToString() {
 			.contains("view=0")
 			.contains(this.author.toString());
 	}
+
+	@Test(expected = NullPointerException.class)
+	public void deserializationWithNullThrowsException1() throws PublicKeyException {
+		new Vote(null, mock(VoteData.class), 1L, mock(ECDSASignature.class), mock(HighQC.class), mock(ECDSASignature.class));
+	}
+
+	@Test(expected = NullPointerException.class)
+	public void deserializationWithNullThrowsException2() throws PublicKeyException {
+		var author = ECKeyPair.generateNew().getPublicKey().getBytes();
+		new Vote(author, null, 1L, mock(ECDSASignature.class), mock(HighQC.class), mock(ECDSASignature.class));
+	}
+
+	@Test(expected = NullPointerException.class)
```

Comment:
Can we use `assertThatThrownBy()` instead as it's what we've been using throughout the code and would ensure that we are testing the NPE on an exact method/constructor call rather than the whole unit test method.

Reply:
There is only one call - constructor, so there are no ambiguity which call might cause an exception.